test/abstractServer.cpp

```diff
@@ -51,6 +51,11 @@ static void EventHandler(mg_connection* conn, int event, void* event_data) {
 
         case MG_EV_HTTP_REQUEST: {
             AbstractServer* server = static_cast<AbstractServer*>(conn->mgr->user_data);
+            auto port_cstr = (char*) malloc(256);
```

Comment:
Please do not use auto where ever possible.

`malloc` is C, consider replacing it with `std::array`.

Reply:
The mongoose webserver mg_conn_addr_to_str requires a char* buffer as input as it's written in C. Would it be better I use stack allocation for the char array?

I'll remove auto for the obvious types.

Reply:
You can get the underlying `char*` from a `std::array` with the `std::array::data()` call: https://en.cppreference.com/w/cpp/container/array/data

In this case it's better tu use stack memory here sine than we do not have to care about free any more.